worker/include/RTC/KeyFrameRequestManager.hpp

```diff
@@ -0,0 +1,98 @@
+#ifndef MS_KEY_FRAME_REQUEST_MANAGER_HPP
+#define MS_KEY_FRAME_REQUEST_MANAGER_HPP
+
+#include "handles/Timer.hpp"
+#include <map>
+
+static uint16_t KeyFrameWaitTime{ 2000 };
+
+namespace RTC
+{
+	class PendingKeyFrameInfo : public Timer::Listener
+	{
+	public:
+		class Listener
+		{
+		public:
+			virtual void OnKeyFrameRequestTimeout(PendingKeyFrameInfo* keyFrameRequestInfo) = 0;
+		};
+
+	public:
+		PendingKeyFrameInfo(Listener* listener, uint32_t ssrc);
+		~PendingKeyFrameInfo();
+
+		uint32_t GetSsrc() const;
+
+		/* Pure virtual methods inherited from Timer::Listener. */
+	public:
+		void OnTimer(Timer* timer) override;
+
+	private:
+		Listener* listener{ nullptr };
+		uint32_t ssrc{ 0 };
+		Timer* timer{ nullptr };
+	};
+
+	class KeyFrameRequestManager : public PendingKeyFrameInfo::Listener
+	{
+	public:
+		class Listener
+		{
+		public:
+			virtual void OnKeyFrameNeeded(uint32_t ssrc) = 0;
+		};
+
+	public:
+		explicit KeyFrameRequestManager(Listener* listener);
+		~KeyFrameRequestManager() = default;
```

Comment:
I suggest removing all those useless destructors that do nothing. I've already removed all non needed virtual methods (child classes do not need to define their methods as virtual at all).